java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -3693,4 +3694,37 @@ public void bitcount(BaseClient client) {
             assertTrue(executionException.getCause() instanceof RequestException);
         }
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void lmpop(BaseClient client) {
+        // setup
+        String key1 = "{key}-1" + UUID.randomUUID();
+        String key2 = "{key}-2" + UUID.randomUUID();
+        long numkeys = 1L;
+        long numkeys2 = 2L;
+        String[] key = {key1};
+        String[] keys = {key2, key1};
+        LmPopOptions lmPopOptions = LmPopOptions.LEFT;
+        long count = 1L;
+        Long arraySize = 5L;
+        String[] lpushArgs = {"one", "two", "three", "four", "five"};
+        Map<String, String[]> expected = Map.of(key1, new String[] {"five"});
+
+        // nothing to be popped
+        assertNull(client.lmpop(numkeys, key, lmPopOptions).get());
+        assertNull(client.lmpop(numkeys, key, lmPopOptions, count).get());
+
+        // pushing to the arrays to be popped
```

Comment:
1. Add more elements to lists and try differernt pop options and count.
2. Add a check with non-lists key(s).
3. Add a cross-slot test (see #1391).